### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by mvanzulli**
This MR: 
- Fixes CI on main
- Docs still fails because I haven't maintainer permissions so I can't add Documenter keys. Be a maintainer would speed up dev operations 😇 
- Set coverage 
- Update README including badges 🍡  . 

CI passing in my fork: https://github.com/mvanzulli/FinancialDerivatives.jl/actions/runs/9050825985/job/24866677183

### CODE REVIEW

**Comment by mvanzulli** [comment]
@ibis: 

 This MR should be ready for review. Not sure if sth else is required on the maintainers or devs with permission side to finalize the set-up. Regarding the CD of pages, I think can be a follow-up, but I believe I need permissions to access to the repo keys.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-runtest@v1
+      - uses: julia-actions/julia-processcoverage@v1
+      - uses: codecov/codecov-action@v3
+        with:
+          files: lcov.info
+          fail_ci_if_error: false
+  docs:
+    name: Documentation
+    runs-on: ubuntu-latest
```
Comment (iblislin) [inline .github/workflows/CI.yml:49]: I just added a key for doc to the `github-pages` environment. 

let's try it out: 

```suggestion
    runs-on: ubuntu-latest
    environment: github-pages
```

Comment (mvanzulli) [reply .github/workflows/CI.yml:49]: In my repo works fine: https://github.com/mvanzulli/FinancialDerivatives.jl/tree/master, maybe after the merge will be trigerred ? Regarding the pages I think Documter key may be needed: https://documenter.juliadocs.org/stable/man/hosting/walkthrough/#Adding-the-Public-Key-to-GitHub-or-Gitea-such-as-Codeberg ?

Comment (iblislin) [reply .github/workflows/CI.yml:49]: yah, I added that.

**Comment by iblislin** [comment]
looks fine 

- https://juliaquant.github.io/FinancialDerivatives.jl/stable/
- https://github.com/JuliaQuant/FinancialDerivatives.jl/actions/runs/9071920106/job/24926539051

:tada: :tada: :tada:

**Comment by mvanzulli** [comment]
> looks fine
> 
>     * https://juliaquant.github.io/FinancialDerivatives.jl/stable/
> 
>     * https://github.com/JuliaQuant/FinancialDerivatives.jl/actions/runs/9071920106/job/24926539051
> 
> 
> 🎉 🎉 🎉

Thanks for your review :100

**Pull request merged**